model/client4.go

```diff
@@ -2282,15 +2296,22 @@ func (c *Client4) UploadFile(data []byte, channelId string, filename string) (*F
 	body := &bytes.Buffer{}
 	writer := multipart.NewWriter(body)
 
-	if part, err := writer.CreateFormFile("files", filename); err != nil {
+	part, err := writer.CreateFormFile("files", filename)
+	if err != nil {
 		return nil, &Response{Error: NewAppError("UploadPostAttachment", "model.client.upload_post_attachment.file.app_error", nil, err.Error(), http.StatusBadRequest)}
-	} else if _, err = io.Copy(part, bytes.NewBuffer(data)); err != nil {
+	}
+
+	_, err = io.Copy(part, bytes.NewBuffer(data))
+	if err != nil {
```

Comment:
Not blocking at all, but I notice this one doesn't use the return value other than `err`, so could stay in the `if` if you want to be a perfectionist :)